petab/simplify.py

```diff
@@ -0,0 +1,106 @@
+"""Functionality for simplifying PEtab problems"""
+from math import nan
+
+import pandas as pd
+
+from . import Problem
+from .C import *  # noqa: F403
+from .lint import lint_problem
+
+__all__ = [
+    "remove_nan_measurements",
+    "remove_unused_observables",
+    "remove_unused_conditions",
+    "simplify_problem",
+    "condition_parameters_to_parameter_table",
+]
+
+
+def remove_nan_measurements(problem: Problem):
+    """Drop any measurements that are NaN"""
+    problem.measurement_df = problem.measurement_df[
+        ~problem.measurement_df[MEASUREMENT].isna()
+    ]
+    problem.measurement_df.reset_index(inplace=True, drop=True)
+
+
+def remove_unused_observables(problem: Problem):
+    """Remove observables that have no measurements"""
+    measured_observables = set(problem.measurement_df[OBSERVABLE_ID].unique())
```

Comment:
Should be equivalent
```suggestion
    measured_observables = set(problem.measurement_df[OBSERVABLE_ID])
```

Reply:
was hoping `unique` was happening in C and might be a bit faster.